### ISSUE TITLE
LilyPond version in Documentation browser

### ISSUE THREAD

**Issue by fedelibre** [frescobaldi/frescobaldi#1352]
How is the version of LilyPond manual pages - in the documentation browser - determined?
I guess it's some fixed value, as I see it keeps showing v2.20 docs despite the fact I've installed the just released v2.22.

While browsing lilypond website within Frescobaldi I was surprised to see the Contributor manual targeted as stable branch. It should exist only as development branch:

    LilyPond — Contributor’s Guide v2.20.0 (stable-branch).

Where is this text coming from?

**Comment by fedelibre** [comment]
>  I guess it's some fixed value, as I see it keeps showing v2.20 docs despite the fact I've installed the just released v2.22.

I should have written "despite the fact the new stable v2.22 has been released", as it's displaying the online documentation.

Anyway, I've added this path in the preferences and it works now:

     http://www.lilypond.org/doc/v2.22

**Comment by fedelibre** [comment]
> While browsing lilypond website within Frescobaldi I was surprised to see the Contributor manual targeted as stable branch. It should exist only as development branch

It seems that the default page shown by the documentation browser is `http://lilypond.org/manuals`, that is the current stable home page for all manuals. This is correct.
However if I click on Development and then Contributor Guide I see this:

![immagine](https://user-images.githubusercontent.com/1644647/203318451-d44afc45-47e8-4711-b432-c4d8bddf076d.png)

Surprised to see the stable branch, but the page exists indeed:
http://lilypond.org/doc/v2.22/Documentation/contributor/index.html

Let's close this issue.

**Comment by fedelibre** [comment]
There's something wrong. Frescobaldi is currently showing 2.22 manuals page instead of 2.24.

**Comment by fedelibre** [comment]
> Anyway, I've added this path in the preferences and it works now:
> 
> ```
>  http://www.lilypond.org/doc/v2.22
> ```

This workaround still works of course (now using v2.24).

If you remove the 2.24 URL from the preferences, it returns again 2.22 pages. This  should exclude a caching issue.

Where this 2.22 comes from then? LilyPond versions added to my Frescobaldi installation are currently 2.24 and 2.25.
There must be a place in Frescobaldi source where (the supposed) current stable is set. I've looked into [frescobaldi_app/lilypondinfo.py](https://github.com/frescobaldi/frescobaldi/blob/master/frescobaldi_app/lilypondinfo.py). AFAICU the documentation version should be the same as the lilypond version set as default, but this is not what's happening.

**Comment by fedelibre** [comment]
I think I've found the culprit. In *Preferences>Documentation browser* you can add a URL. If you start typing `http://` you'll see a few URLs coming up. The most useful are the following:

![immagine](https://user-images.githubusercontent.com/1644647/212567194-cf535016-15c7-476d-9d5e-e14518fc6a19.png)

In current Frescobaldi, stable translates to 2.22 instead of 2.24 and unstable to 2.23.82 instead of 2.25. However, those links are working fine in a browser. So I guess that QWebEngineView should use some cache clearing mechanism.

@dliessi @PeterBjuhr do you have any experience with this Qt stuff?

**Comment by fedelibre** [comment]
I guess this page can help:
https://www.riverbankcomputing.com/static/Docs/PyQt5/api/qtwebenginewidgets/qwebengineprofile.html

**Comment by PeterBjuhr** [comment]
I don't have any experience of caching of web pages in QT. But in your link there's the method `clearHttpCache()`. Could be worth a try!?

**Comment by PeterBjuhr** [comment]
When I try the links in my browser:
http://lilypond.org/doc/stable/ goes to http://lilypond.org/manuals which shows "Manuals for LilyPond 2.24.0"
and
http://lilypond.org/doc/latest/ goes to http://lilypond.org/doc/v2.23/Documentation/web/index.html

**Comment by fedelibre** [comment]
You are right. The /latest link should be fixed in lilypond right [here](https://gitlab.com/lilypond/lilypond/-/blob/master/Documentation/webserver/lilypond.org.htaccess#L48). I'll send a patch later.

If you want to give a try witth clearHttpCache go ahead.

**Comment by PeterBjuhr** [comment]
`clearHttpCache` didn't help. But in the link you sent I see that the stable matching also seems to redirect to the old version. That would have explained everything except it do work for stable in the browser.

**Comment by PeterBjuhr** [comment]
I see now what is happening.

The url in Frescobaldi (when I debug it) is: http://lilypond.org/doc/stable/Documentation/web/manuals not just http://lilypond.org/doc/stable/

which means the latest redirect match is 
`RedirectMatch ^(/Documentation.*)$ /doc/v2.22$1`

**Comment by fedelibre** [comment]
Here's the MR:
https://gitlab.com/lilypond/lilypond/-/merge_requests/1815

I guess we can close this issue again as soon the MR has been merged.

Thanks for your help!

**Comment by fedelibre** [comment]
What you found out seems to be what Urs reported in #1163

**Comment by PeterBjuhr** [comment]
The MR looks good as far as I can tell. And should solve this issue. But I will take a closer look at what Urs is writing.
